Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:20.04
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get -qq -y update \
+    && apt-get -qq -y install python3-pip python3-dev python3-pil
+
+RUN apt-get -qq -y install pkg-config libicu-dev
+RUN pip install --no-binary=:pyicu: pyicu
+
+# Environment variables used for the SQLAlchemy 2.0 migration
+ENV SQLALCHEMY_WARN_20='1' \
+    PYTHONWARNINGS='always'
+
+COPY . /ftmstore
+WORKDIR /ftmstore
+
+RUN pip3 install --no-cache-dir followthemoney SQLAlchemy postgres pytest pyicu
```

Comment:
Ah, sorry, my wording was unclear.

Ingas suggesting to install the dependencies by referring `setup.py` with `pip install -e .[postgresql]` rather than hardcoding them in the Dockerfile.

But as I'm not super knowledgeable about all the package tooling in the Python ecosystem, there may be something I'm overlooking why that's not possible!

Reply:
I guess we are hardcoding the dependencies because there's no easy way of saying `pip install dependencies from setup.py, but not the project itself`. The typical workaround here would be to have a `requirements.txt` file and then `setup.py` to read that (see for instance https://stackoverflow.com/questions/26900328/install-dependencies-from-setup-py). I would advise doing that because it would help clear up the project structure, especially for local development.

Reply:
Okay, I don’t have any opinion on this at all, just trying to learn something, so I have a follow-up question:

When I have a package in `/foo` and I run `pip install -e /foo` it installs the package (and also its requirements). So if I have another project `bar` that uses `foo` as a dependency there could be a conflict. I thought that’s the reason why we use virtual environments?

So the alternative is to keep a separate `requirements.txt` for development. But then you start maintaining the same list of dependencies in `requirements.txt` and `setup.py`? 🤯

Reply:
The virtual environment doesn't protect you from transitive dependency hell, unfortunately. But stating your requirements in a txt file makes it easier to create a virtualenv (or provision a Docker image). The next step here would be to use a lock file as well.

You don't need to have the dependencies listed twice, because you can have `setup.py` read the `requirements.txt` file.

Reply:
I created a `requirements.txt` file, added the dependencies. I kept the `extras_require` in `setup.py` as per [best practices](https://stackoverflow.com/questions/41268863/what-is-the-difference-between-extras-require-and-install-requires-in-se). Also, I checked the `setup.py` for aleph and it seems like [it doesn't read the requitements.txt file](https://github.com/alephdata/aleph/blob/main/setup.py#L20) so I didn't either. This is up for change if we feel we should implement it differently.

Reply:
Thanks, @catileptic ! I feel like we don't `pip install aleph`, so it's not that important, whereas we do depend on `followthemoney-store` in a few places, so it would be great for `setup.py` and `requirements.txt` not to drift apart. Otherwise we're running the risk of developing against something different than what we're installing.